Sources/PushNotifications.swift

```diff
@@ -135,18 +144,27 @@ import Foundation
         }
 
         let persistenceService: InterestPersistable = PersistenceService(service: UserDefaults(suiteName: "PushNotifications")!)
-        if persistenceService.persist(interest: interest) {
-            serialQueue.async {
-                guard
-                    let deviceId = Device.getDeviceId(),
-                    let instanceId = Instance.getInstanceId(),
-                    let url = URL(string: "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances/\(instanceId)/devices/apns/\(deviceId)/interests/\(interest)")
-                    else { return }
 
-                let networkService: PushNotificationsNetworkable = NetworkService(url: url, session: self.session)
-                networkService.subscribe(completion: { (_, _) in
+        if Device.idAlreadyPresent() {
+            if persistenceService.persist(interest: interest) {
+                networkQueue.async {
+                    guard
+                        let deviceId = Device.getDeviceId(),
+                        let instanceId = Instance.getInstanceId(),
+                        let url = URL(string: "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances/\(instanceId)/devices/apns/\(deviceId)/interests/\(interest)")
+                        else { return }
+
+                    let networkService: PushNotificationsNetworkable = NetworkService(url: url, session: self.session)
+                    networkService.subscribe(completion: { (_, _) in
+                        completion()
+                    })
+                }
+        } else {
+                persistenceService.persist(interest: interest)
+                localStorageQueue.async {
+                    persistenceService.persist(interest: interest)
                     completion()
```

Comment:
I was unsure about this `completion` handler, now I'm even more unsure.

Sometimes it will get called when the server has confirmed receiving the request, sometimes it will get called when we write to disk.

I think we should remove it as we don't need this extra complexity.